examples/prometheus/prometheus.yaml

```diff
@@ -133,24 +142,92 @@ objects:
           imagePullPolicy: IfNotPresent
           volumeMounts:
           - mountPath: /etc/prometheus
-            name: config-volume
+            name: prometheus-config
           - mountPath: /prometheus
-            name: data-volume
+            name: prometheus-data
+
+        # Deploy alertmanager behind prometheus-alert-buffer behind an oauth proxy
```

Comment:
You can support multiple upstreams if they don't overlap in URLs.  But I don't know why two proxies is an issue - you can target services to any pod you want, in any port you want.